### PR TITLE
Edited DeveloperGuide, PPP

### LINKED ISSUE #177: docs: update Developer Guide with product scope, NFR, glossary
- Included more NFRs, and Glossary definitions. 
- Added more explanations on target user profile, and value proposition in product scope.

### LINKED ISSUE #178: docs: updated PPP
updated PPP under `JavonTeo.md`

### PR DESCRIPTION
**Pull request by JavonTeo**
This resolves issues #177 and #178.

### CODE REVIEW

File: docs/team/javonteo.md [hunk 1]
```diff
@@ -1,38 +1,41 @@
 ---
-layout: page
-title: Javon Teo's Project Portfolio Page
+Javon Teo's Project Portfolio Page
```
Comment (Junyi00) [inline docs/team/javonteo.md]: I believe the removed lines are necessary for our project website to render your portfolio page correctly

**Review by Junyi00 — CHANGES_REQUESTED** [review]
Just a minor change to ensure your portfolio is rendered correctly in our project page!

**Review by Junyi00 — APPROVED** [review]
LGTM!

File: docs/team/javonteo.md (hunk 1 above)
Comment (JavonTeo) [reply docs/team/javonteo.md]: Resolved in the next commit

**Pull request merged**